app-admin/pinkybar/pinkybar-9999.ebuild

```diff
@@ -5,136 +5,157 @@ EAPI=6
+	lisp? ( dev-lisp/ecls )
 	lua? ( dev-lang/lua )
-	ruby? ( dev-lang/ruby )
+	mpd? ( media-sound/mpd media-libs/libmpdclient )
+	ncurses? ( sys-libs/ncurses )
+	net? ( sys-apps/iproute2 )
+	pci? ( sys-apps/pciutils )
+	python2? ( ${PYTHON_DEPS} )
 	r? ( dev-lang/R )
-	lisp? ( dev-lisp/ecls )
+	ruby? ( dev-lang/ruby )
+	sensors? ( sys-apps/lm_sensors )
 	slang? ( sys-libs/slang )
+	smartemp? ( sys-apps/smartmontools )
 	tcl? ( dev-lang/tcl )
+	weather? ( net-misc/curl app-arch/gzip )
+	x11? ( x11-libs/libX11 )
+
 "
 REQUIRED_USE="
-	x11? ( !ncurses )
-	ncurses? ( !x11 )
 	drivetemp? ( !smartemp )
 	drivetemp-light? ( !smartemp )
+	ncurses? ( !x11 )
 	smartemp? ( !drivetemp !drivetemp-light )
+	x11? ( !ncurses )
 "
 
-pkg_setup() {
-	if use weather
-	then
-		einfo 'Currently, the weather USE flag will default to London,uk'
-		einfo 'To specify other country and town youll have to supply them as variable.'
-		einfo 'Here is how: # TWN="London,uk" USE="weather" emerge -a ...'
-	fi
-}
-
 src_prepare() {
 	default
 
-	einfo 'Generating Makefiles'
+	elog 'Generating Makefiles'
 	perl set.pl 'gentoo' || die
-	autoreconf -if || die
+	eautoreconf -if
 }
 
 src_configure() {
-	export PKG_CONFIG_PATH=/usr/bin/pkg-config
-
-	econf \
-		$(use_with x11) \
-		$(use_with lua) \
-		$(use_with ruby) \
-		$(use_with r) \
-		$(use_with lisp) \
-		$(use_with slang) \
-		$(use_with tcl) \
-		$(use_with cpp) \
-		$(use_with x11) \
-		$(use_with alsa) \
-		$(use_with net) \
-		$(use_with libnl) \
-		$(use_with pci) \
-		$(use_with dvd) \
-		$(use_with sensors) \
-		$(use_with ncurses) \
-		$(use_with colours) \
-		$(use_with weather) \
-		$(use_with mpd) \
-		$(use_with drivetemp) \
-		$(use_with drivetemp-light) \
-		$(use_with smartemp) \
-		$(use_with perl) \
-		$(use_with python2) \
-		perl_script='/usr/share/pinkysc/pinky.pl' \
-		python_script='/usr/share/pinkysc/pinky.py' \
-		api_town="${TWN:-London,uk}" \
-		api_key='28459ae16e4b3a7e5628ff21f4907b6f' \
+
+	CONFIGURE_OPTS=(
+		$(use_with alsa)
+		$(use_with colours)
+		$(use_with cpp)
+		$(use_with drivetemp)
+		$(use_with drivetemp-light)
+		$(use_with dvd)
+		$(use_with libnl)
+		$(use_with lisp)
+		$(use_with lua)
+		$(use_with mpd)
+		$(use_with ncurses)
+		$(use_with net)
+		$(use_with pci)
+		$(use_with perl)
+		$(use_with python2)
+		$(use_with r)
+		$(use_with ruby)
+		$(use_with sensors)
+		$(use_with slang)
+		$(use_with smartemp)
+		$(use_with tcl)
+		$(use_with weather)
+		$(use_with x11)
+		api_key='28459ae16e4b3a7e5628ff21f4907b6f'
 		icons='/usr/share/icons/xbm_icons'
+	)
+	econf "${CONFIGURE_OPTS[@]}"
 }
 
 src_compile() {
-	emake 'all' || die
+	emake
 }
 
 src_install() {
 	if use colours && ! use x11 && ! use ncurses
 	then
-		insinto /usr/share/icons/xbm_icons
-		doins "${S}"/extra/xbm_icons/*
+		insinto /usr/share/icons
+		doins -r /extra/xbm_icons
 	fi
 
-	if use perl || use python2 || use lua || \
-		use r || use ruby || use tcl \
-		use slang || use lisp
+	if use lisp
```

Comment:
Is there a way to use switch `case` in here to just `doins` instead using `if` multiple times ?

Reply:
Nope. Just use the short form:

    use foo && doins ...